tests/name_constraints/generate.py

```diff
@@ -0,0 +1,282 @@
+from cryptography import x509
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.primitives.serialization import Encoding
+from cryptography.hazmat.backends import default_backend
+from cryptography.x509.oid import NameOID
+import ipaddress
+import datetime
+
+ISSUER_PRIVATE_KEY = rsa.generate_private_key(
+    public_exponent=65537,
+    key_size=2048,
+    backend=default_backend()
+)
+ISSUER_PUBLIC_KEY = ISSUER_PRIVATE_KEY.public_key()
+
+NOT_BEFORE = datetime.datetime.fromtimestamp(0x1fedf00d - 30)
+NOT_AFTER = datetime.datetime.fromtimestamp(0x1fedf00d + 30)
+
+def name_constraints_test(test_name,
```

Comment:
Maybe add some comments here to explain these arguments? I also got a little confused by the argument order, for example I would suggest that `valid_names` and `invalid_names` come last (since they end up making up the assertions) and `sans` come right after `extra_subject_names`.

Some general comments about the approach here (Python script generates certificates and a Rust test case for them) would be nice, too.